### PR TITLE
use passport-nevermined

### PR DESCRIPTION
**Pull request by r-marques**
## Description

- Use passport-nevermined to handle authentication
- Refactor auth module
- Cleanup no longer needed code
- move jose to dev dependencies

## Is this PR related with an open issue?

Closes https://github.com/nevermined-io/internal/issues/355

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [ ] Follows the code style of this project.
- [ ] Tests Cover Changes
- [ ] Documentation

### CODE REVIEW

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -27,7 +28,32 @@ export class AuthController {
...
   @Public()
   login(@Body() clientAssertion: ClientAssertionDto): Promise<LoginDto> {
-    return this.authService.validateClaim(clientAssertion.client_assertion_type, clientAssertion.client_assertion)
+    return this.authService.validateClaim(
+      clientAssertion.client_assertion_type,
+      clientAssertion.client_assertion,
+    )
+  }
+
+  @Post('login2')
```
Comment (aaitor) [inline src/auth/auth.controller.ts:37]: maybe a better name than `login2`

File: src/auth/auth.controller.ts [hunk 2]
```diff
@@ -27,7 +28,32 @@ export class AuthController {
...
+    description: 'The access_token',
+    type: LoginDto,
+  })
+  @ApiResponse({
+    status: 401,
+    description: 'Unauthorized access',
+  })
+  @Public()
+  @UseGuards(NeverminedGuard)
+  async login2(@Req() req, @Body() _clientAssertion: ClientAssertionDto): Promise<LoginDto> {
```
Comment (aaitor) [inline src/auth/auth.controller.ts:53]: same

File: src/auth/auth.controller.ts [hunk 3]
```diff
@@ -27,7 +28,32 @@ export class AuthController {
...
+    console.log(req.headers)
+    console.log('calling login2', req.user)
```
Comment (aaitor) [inline src/auth/auth.controller.ts:55]: same

**Review by aaitor — APPROVED** [review]
lgtm, I would rename the `login` method

**Review by dancespiele — APPROVED** [review]

**Pull request merged**